test/jdk/javax/swing/text/html/HTMLEditorKit/bug4238223.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 1999, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.Reader;
+import java.io.StringReader;
+import javax.swing.text.MutableAttributeSet;
+import javax.swing.text.html.HTML;
+import javax.swing.text.html.HTMLEditorKit;
+import javax.swing.text.html.parser.ParserDelegator;
+
+/*
+ * @test
+ * @bug 4238223
+ * @summary Tests that HTMLEditorKit.ParserCallback methods receive
+ *          correct `pos' argument.
+ */
+
+public class bug4238223 {
+
+    public static void main(String[] argv) throws Exception {
+        TestParser parser = new TestParser();
+        String testHTML = "<HTML><HEAD><TITLE>Text</TITLE></HEAD>" +
+                "<BODY><WRONGTAG>Simple text<!--comment--></BODY></HTML>";
+        parser.parse(testHTML);
+    }
+
+    static class TestCallback extends HTMLEditorKit.ParserCallback {
+        String commentData = "comment";
+        int commentIndex = 65;
+
+        public void handleComment(char[] data, int pos) {
+            if (! (new String(data)).equals(commentData) ||
+                    pos != commentIndex) {
+
+                throw new RuntimeException("handleComment failed");
+            }
+        }
+
+        HTML.Tag[] endTags = {HTML.Tag.TITLE, HTML.Tag.HEAD,
+                HTML.Tag.BODY, HTML.Tag.HTML};
+        int[] endTagPositions = {23, 31, 79, 86};
+        int endTagIndex = 0;
+        public void handleEndTag(HTML.Tag tag, int pos) {
+            if (! tag.equals(endTags[endTagIndex]) ||
+                    pos != endTagPositions[endTagIndex]) {
```

Comment:
```suggestion
            if (!tag.equals(endTags[endTagIndex])
                    || pos != endTagPositions[endTagIndex]) {
```
No space after `!`, wrap before the operator.